hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java

```diff
@@ -1401,7 +1401,7 @@ public OfflineRegionResponse offlineRegion(RpcController controller,
         master.cpHost.preRegionOffline(hri);
       }
       LOG.info(master.getClientIdAuditPrefix() + " offline " + hri.getRegionNameAsString());
-      master.assignmentManager.regionOffline(hri);
+      master.assignmentManager.regionOffline(hri, true);
```

Comment:
Shouldn't we check for the new state if its `OFFLINE` before forcing it to remove ?

Reply:
Or I guess all the admin.offline call has final state as OFFLINE.

Reply:
Region state is actually not updated by fsck. This RPC is used by fsck only, so the region should always be forced offline here.